index.html

```diff
@@ -367,6 +370,9 @@ <h3>Requirements</h3>
                  rendered.
    N28           TBD: restrictions on the application so as to
                  prevent unauthorized recording of the session.
+   N29           The application must be able to request user consent
+                 for one-way media and data only use cases in a
+                 non-discriminating way.
```

Comment:
What does "non-discriminating way" mean?

Reply:
This can easily be misinterpreted as one-way media and data should not be allowed without user consent. This is about better connections (by exposing IP addresses), but it's not clear what the user is consenting to

Reply:
> What does "non-discriminating way" mean?

The intent is to ensure that there is at least one use case neutral way to request user consent. Maybe I should rephrase? Any suggestions?

Reply:
> This can easily be misinterpreted as one-way media and data should not be allowed without user consent. This is about better connections (by exposing IP addresses), but it's not clear what the user is consenting to

I don't understand that. When we're talking about consent, it usually means the form of *consent* as described by the IP handling draft which is what this is targeting. Should I clarify this? Suggestions?

Reply:
Agree with @henbos, this needs to talk specifically about IP address enumeration, not the use cases (most of which don't need IP address enumeration)

Reply:
The *requirement* here is that the API should prevent abusive and privacy-violating usage; whether that is done by requesting consent or by other means is not a requirement, it's an implementation suggestion.

Reply:
Note that the cited use cases work fine today in most environments, even without additional permission grants, so additional detail is needed to flesh out the actual needs here.

Reply:
How about "one way media and data communication between users inside the same LAN"

Reply:
> Note that the cited use cases work fine today in most environments, even without additional permission grants, so additional detail is needed to flesh out the actual needs here.

That is not entirely correct, so I'll summarise some issues that exist for use cases benefiting highly from direct connections, much more so than audio/video use cases would (e.g. file transfer, ideally within the same LAN):

1. Multi-segment (e.g. corporate) LANs tend to filter multicast, so obfuscated (mDNS) host candidates do not help.
2. NAT hairpinning is not supported in many routers (e.g. Telekom), so srflx candidates do not help.
3. Many cable providers use DS-lite, round-tripping NAT hairpinning over the DS-lite gateway (e.g. Unitymedia), so srflx candidates do not help (again).

---

I've made a suggestion below on how to rephrase this but got not feedback so far, see https://github.com/w3c/webrtc-nv-use-cases/pull/14#issuecomment-586904662

Reply:
To clarify what I mean with *non-discriminating*: https://github.com/w3c/webrtc-pc/pull/2175 is a good example on how to mitigate that. It just says there are two camps: Either you have allowance for a "direct connection" or you don't and that permission is generally use case independent. How the permission can be achieved is then up to the UA. The only requirement is that it can **also** be obtained *non-discriminately* i.e. not gated by the `getUserMedia` (aka *take a selfie before sharing files*). That PR also talks about what other variants there may be.

> The _requirement_ here is that the API should prevent abusive and privacy-violating usage; whether that is done by requesting consent or by other means is not a requirement, it's an implementation suggestion.

If I understand correctly that just means that _requesting user consent_ is not part of the spec, so it should not be used in a requirement? That would make sense to me. See my below suggestion (https://github.com/w3c/webrtc-nv-use-cases/pull/14#issuecomment-586904662) on a rephrase attempt.

Reply:
I think that "between browsers inside the same LAN" is the key qualifier here, as illustrated by scenarios 1-3 in https://github.com/w3c/webrtc-nv-use-cases/pull/14#discussion_r555594713. This only affects cases where both endpoints are browsers because native applications don't need to use mDNS.

I prefer this sort of framing versus https://github.com/w3c/webrtc-nv-use-cases/pull/14#issuecomment-586904662 because it gets at the why rather than the what.

Reply:
Okay, perfectly fine with that as it also points out more clearly what concrete use cases it targets. I'll /cc @steely-glint just in case. Justin, could you make a concrete proposal that would work for you? I'll happily update the PR with your framing.

Reply:
"N29: Support for one-way media and data applications between browsers inside the same LAN", which basically reduces to finding an alternative to mDNS for this environment.

You could also call out "NXX: Ability for one-way media and data applications to control the network interface used by ICE" as a separate requirement, which may point at a different solution. This particular problem largely reduces to a special case consideration to route real-time traffic directly when a VPN is in use, and may be solvable via its own targeted solution.

Reply:
I'm reasonably happy with "N29: Support for one-way media and data applications between browsers inside the same LAN" but I have 2 minor issues. 
1. I'd like to emphasize that the goal is the media should stay on that LAN, otherwise TURN fits the bill.
2. LAN is a bit vague. We may be talking about anything from a local wifi to a whole corp network. (I'm thinking of the multisegment star network that traditionally covers the whole 4,400 acres of the burning man site - which isn't obviously a LAN, or the 15000 participant network that CCC spin up at congress.)

I don't agree with your framing of NXX as a VPN problem - it also addresses outages - so lets discuss that in a different issue.

Reply:
yea, or a corp network that uses MPLS and site-site VPN to make it all look like 1 big LAN. Is there a standard name for a "continuously virtual connected not NATed network" ?

Reply:
perhaps we could add 'p2p' to 1), e.g., "N29: Support for one-way media and data applications to connect P2P to other browsers inside the same LAN". Definition of what exactly is meant by LAN is probably better left as sub-bullets.